### PR TITLE
cli: move workload & seedshare owner key handling into manifest

### PR DESCRIPTION
**Pull request by Freax13**
This PR removes code related to workload and seedshare key handling from the CLI subcommands and consolidates it into the manifest package. This PR also adds some unit tests for the new functions.

### CODE REVIEW

File: cli/cmd/generate.go [hunk 1]
```diff
@@ -415,21 +393,11 @@ func addSeedshareOwnerKeyToManifest(manifst *manifest.Manifest, keyPath string)
...
-	switch block.Type {
-	case "RSA PRIVATE KEY":
-		privateKey, err := x509.ParsePKCS1PrivateKey(block.Bytes)
-		if err != nil {
-			return fmt.Errorf("parsing RSA private key: %w", err)
-		}
-		publicKey = manifest.MarshalSeedShareOwnerKey(&privateKey.PublicKey)
-	default:
-		return fmt.Errorf("unsupported PEM block type: %s", block.Type)
+	privateKey, err := manifest.ParseSeedshareOwnerPrivateKey(keyData)
```
Comment (burgerdev) [inline cli/cmd/generate.go:396]: Could this be more similar to the workload owner key function, like `ExtractSeedshareOwnerPublicKey`?

File: internal/manifest/crypto_test.go [hunk 2]
```diff
@@ -96,3 +96,29 @@ func newKey(t *testing.T, bits int) *rsa.PrivateKey {
...
+}
+
+func TestWorkloadOwnerKeyParseMarshal(t *testing.T) {
+	keyData, err := NewWorkloadOwnerKey()
+	require.NoError(t, err)
+
+	privateKey, err := ParseWorkloadOwnerPrivateKey(keyData)
+	require.NoError(t, err)
+
+	HashWorkloadOwnerKey(&privateKey.PublicKey)
```
Comment (burgerdev) [inline internal/manifest/crypto_test.go:120]: Does this need testing, e.g. `size > 0`?

File: internal/manifest/crypto_test.go [hunk 3]
```diff
@@ -96,3 +96,29 @@ func newKey(t *testing.T, bits int) *rsa.PrivateKey {
...
+
+	_, err = ExtractWorkloadOwnerPublicKey(keyData)
```
Comment (burgerdev) [inline internal/manifest/crypto_test.go:122]: Could we add a test for the `PUBLIC KEY` case?

**Review by burgerdev — COMMENTED** [review]
Nice, thanks!

File: internal/manifest/crypto_test.go (hunk 3 above)
Comment (Freax13) [reply internal/manifest/crypto_test.go]: Sure.

File: internal/manifest/crypto_test.go (hunk 2 above)
Comment (Freax13) [reply internal/manifest/crypto_test.go]: My main intention with putting this in the test was to make sure the function won't panic, but sure, we can also test for the length.

File: internal/manifest/crypto_test.go [hunk 4]
```diff
@@ -96,3 +98,43 @@ func newKey(t *testing.T, bits int) *rsa.PrivateKey {
...
+	keyData, err := NewSeedShareOwnerPrivateKey()
+	require.NoError(t, err)
+
+	key, err := ParseSeedshareOwnerPrivateKey(keyData)
+	require.NoError(t, err)
+
+	pubHexStr := MarshalSeedShareOwnerKey(&key.PublicKey)
+
+	_, err = ParseSeedShareOwnerKey(pubHexStr)
+	require.NoError(t, err)
```
Comment (burgerdev) [inline internal/manifest/crypto_test.go:112]: This looks like a leftover (the function is not defined?).

File: internal/manifest/crypto_test.go [hunk 5]
```diff
@@ -96,3 +98,43 @@ func newKey(t *testing.T, bits int) *rsa.PrivateKey {
...
+
+	publicKeyHexStr, err := ExtractSeedshareOwnerPublicKey(keyData)
+	require.NoError(t, err)
+
+	publicKeyBytes, err := publicKeyHexStr.Bytes()
+	require.NoError(t, err)
+	publicKeyPem := pem.EncodeToMemory(&pem.Block{Type: "PUBLIC KEY", Bytes: publicKeyBytes})
+	_, err = ExtractSeedshareOwnerPublicKey(publicKeyPem)
```
Comment (burgerdev) [inline internal/manifest/crypto_test.go:120]: Nit: we could also test that the extracted public key is the same in both cases.

**Review by burgerdev — APPROVED** [review]

File: internal/manifest/crypto_test.go (hunk 4 above)
Comment (burgerdev) [reply internal/manifest/crypto_test.go:112]: Nevermind, missed that it already existed.

Comment (Freax13) [reply internal/manifest/crypto_test.go:112]: The function is defined [here](https://github.com/edgelesssys/contrast/blob/81bbc6bf5acd92a1230abfffd55e25a617ae08bf/internal/manifest/crypto.go#L75).

**Pull request merged**